src/ORM/UnionList.php

```diff
@@ -0,0 +1,261 @@
+<?php
+
+namespace SilverStripe\ORM;
+
+use SilverStripe\Dev\Debug;
+use SilverStripe\View\ViewableData;
+use LogicException;
+
+/**
+ * A list object that wraps around other SS_List instances.
+ * Allows iteration over multiple collections in one loop.
+ */
+class UnionList extends ViewableData implements SS_List
+{
+    /**
+     * An array of SS_List instances
+     *
+     * @var array
+     */
+    protected $lists = array();
+
+    /**
+     * Create a new UnionList.
+     *
+     * @param array $lists - Lists to hold
+     */
+    public function __construct(array $lists)
+    {
+        foreach ($lists as $list) {
+            $this->lists[] = clone $list;
+        }
+        parent::__construct();
+    }
+
+    /**
+     * Return the total number of items in each SS_List.
+     *
+     * @return int
+     */
+    public function count()
+    {
+        $count = 0;
+        foreach ($this->lists as $list) {
+            $count += $list->count();
+        }
+        return $count;
+    }
+
+     /**
+     * Returns true if one of the SS_List's have an item.
+     *
+     * @return bool
+     */
+    public function exists()
+    {
+        foreach ($this->lists as $list) {
+            if ($list instanceof ArrayList) {
+                if ($list->exists()) {
+                    return true;
+                }
+            } elseif (method_exists($list, 'exists') && $list->exists()) {
```

Comment:
This condition can be used alone, we don't need to check if the list is an `instanceof ArrayList` above because this check will handle that situation